File: backend/routes/genre.go
package routes

import (
	"FilmFindr/controller"
	"FilmFindr/helpers"
	"FilmFindr/middleware"
	"FilmFindr/service"

	"github.com/gin-gonic/gin"
)

func GenreRoute(router *gin.Engine, genreController controller.GenreController, jwtService service.JWTService) {
	genre := router.Group("/api/v1/genres")
	{
		genre.GET("/", middleware.Authenticate(jwtService), middleware.AuthorizeRole(helpers.ENUM_ROLE_ADMIN), genreController.GetAllGenre)
		genre.POST("/", middleware.Authenticate(jwtService), middleware.AuthorizeRole(helpers.ENUM_ROLE_ADMIN), genreController.CreateGenre)
		genre.DELETE("/:id", middleware.Authenticate(jwtService), middleware.AuthorizeRole(helpers.ENUM_ROLE_ADMIN), genreController.DeleteGenre)
	}
}

File: frontend/src/service/genre.js
import axios from "axios"
import { BASE_API_URL } from "../utils/constant";

const BASE_URL = `${BASE_API_URL}/genres`;

export const getAllGenre = async () => {
    const response = await axios.get(`${BASE_URL}/`, {
        withCredentials: true,
    });
    return response;
}

export const createGenre = async (data) => {
    const response = await axios.post(`${BASE_URL}/`, data, {
        withCredentials: true,
    });
    return response;
}

export const deleteGenre = async (id) => {
    const response = await axios.delete(`${BASE_URL}/${id}`, {
        withCredentials: true,
    });
    return response;
}